pelican/__init__.py

```diff
@@ -390,6 +394,19 @@ def main():
 
     try:
         pelican, settings = get_instance(args)
+
+        if args.dump_settings:
+            from pprint import pprint
+
+            # 'pelican --dump-settings': dump_settings = [None]
+            if args.dump_settings[0] is None:
+                pprint(settings)
+
+            # 'pelican --dump-settings SETTING': dump_settings = [SETTING]
+            else:
+                pprint(settings[args.dump_settings[0]])
```

Comment:
A nice warning/error instead of a `KeyError` exception would be nice if the given argument is not in the settings dict.